11_developer_primer/3_developing_for_dynamo/advanced-dynamo-node-customisation.md

```diff
@@ -180,7 +180,181 @@ public static void ProcessFile(string filePath)
+#### Step 1: Import the Required Namespace <a href="#step-1-import-the-required-namespace" id="step-1-import-the-required-namespace"></a>
+
+`OnLogWarningMessage` is part of the `DynamoServices` namespace, so begin by adding this to your project file.
+
+#### Step 2: Identify When to Log Information <a href="#step-2-identify-when-to-log-information" id="step-2-identify-when-to-log-information"></a>
+
+Before adding an info message, think about the purpose of your method:
+
+* What information would be useful to confirm after an action is completed?
+* Are there key steps or milestones within the method that users might want to know about?
+
+Examples of useful confirmations:
+
+* **Completion messages** (e.g., when a grid or model is fully created).
+* **Details of processed data** (e.g., 10 items processed successfully).
+* **Execution summaries** (e.g., parameters used in the process).
+
+#### Step 3: Use `OnLogInfoMessage` to Log Informational Messages <a href="#step-3-use-onloginfomessage-to-log-informational-message" id="step-3-use-onloginfomessage-to-log-informational-message"></a>
+
+Place `OnLogInfoMessage` calls at meaningful points in your method. When a key step or completion occurs, log an informational message to update the user on what happened.
+
+### Syntax for `OnLogInfoMessage` <a href="#syntax-for-onloginfomessage" id="syntax-for-onloginfomessage"></a>
+
+```
+LogWarningMessageEvents.OnLogInfoMessage("Your info message here.");
+```
+
+### Example Implementations of `OnLogInfoMessage` <a href="#example-implementations-of-onloginfomessage" id="example-implementations-of-onloginfomessage"></a>
+
+Here are different scenarios to demonstrate using `OnLogInfoMessage` in your Zero Touch nodes.
+
+#### Example 1: Validating Numeric Inputs <a href="#example-1-validating-numeric-inputs" id="example-1-validating-numeric-inputs"></a>
+
+In this example, well build upon the custom node created in the previous **Zero-Touch Case Study - Grid Node;** A Method named `RectangularGrid` that generates a grid of rectangles based on `xCount` and `yCount` inputs. We will walk through testing If an input is invalid, and then using `OnLogInfoMessage` to provide info after the node has completed its run.
+
+![OnLogInfoMessage Example 1](images/onloginfomessage-example-1.png)
+
+###### Using `OnLogInfoMessage` for Input Validation <a href="#using-onloginfomessage-for-unput-validation" id="using-onloginfomessage-for-unput-validation"></a>
+
+When generating a grid based on `xCount` and `yCount`. After generating the grid, you want to confirm its creation by logging an informational message with the grids dimensions.
+
+```
+public static List<Rectangle> CreateGrid(int xCount, int yCount)
+{
+    // Check if xCount and yCount are positive
+    if (xCount <= 0 || yCount <= 0)
+    {
+        LogWarningMessageEvents.OnLogWarningMessage("Grid count values must be positive integers.");
```

Comment:
Are we supposed to update the code snippet here as well?

Reply:
I have updated the code snippet, nice spot